### PR TITLE
CIV-13071 Claimant dashboard notification defendant requests more time

### PR DESCRIPTION
**Pull request by paul-pearson**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/CIV-13071

### Change description ###

Claimant dashboard notification defendant requests more time

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/civil/callback/CaseEvent.java [hunk 1]
```diff
@@ -369,7 +369,10 @@ public enum CaseEvent {
     CREATE_DEFENDANT_DASHBOARD_NOTIFICATION_FOR_CLAIMANT_RESPONSE(CAMUNDA),
     CREATE_DASHBOARD_NOTIFICATION_FOR_CCJ_REQUEST_FOR_RESPONDENT1(CAMUNDA),
     GENERATE_DASHBOARD_NOTIFICATION_CLAIM_ISSUE_HWF_CLAIMANT1(CAMUNDA),
-    CREATE_DASHBOARD_NOTIFICATION_FOR_CCJ_REQUEST_FOR_APPLICANT1(CAMUNDA);
+    CREATE_DASHBOARD_NOTIFICATION_FOR_CCJ_REQUEST_FOR_APPLICANT1(CAMUNDA),
+    CREATE_CLAIMANT_DASHBOARD_NOTIFICATION_FOR_DEFENDANT_RESPONSE(CAMUNDA),
```
Comment (paul-pearson) [inline src/main/java/uk/gov/hmcts/reform/civil/callback/CaseEvent.java:373]: Check these

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/dashboardnotifications/defendant/ResponseDeadlineExtendedDashboardNotificationHandler.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+    @Override
+    public List<CaseEvent> handledEvents() {
+        return EVENTS;
+    }
+
+    private CallbackResponse configureScenarioForDefendantResponse(CallbackParams callbackParams) {
+        CaseData caseData = callbackParams.getCaseData();
+        String authToken = callbackParams.getParams().get(BEARER_TOKEN).toString();
+        if (caseData.isRespondent1NotRepresented()) {
```
Comment (GorilaNaranja) [inline src/main/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/dashboardnotifications/defendant/ResponseDeadlineExtendedDashboardNotificationHandler.java:52]: As the only change depending on the condition is the scenario I think is better to a getScenario method that return a scenario depending on the conditions like we are doing here:
https://github.com/hmcts/civil-service/pull/4235/files

So we will only have one recordScenario

**Review by laarmada — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/reform/civil/callback/CaseEvent.java (hunk 1 above)
Comment (paul-pearson) [reply src/main/java/uk/gov/hmcts/reform/civil/callback/CaseEvent.java]: Removed

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/dashboardnotifications/defendant/ResponseDeadlineExtendedDashboardNotificationHandler.java (hunk 2 above)
Comment (paul-pearson) [reply src/main/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/dashboardnotifications/defendant/ResponseDeadlineExtendedDashboardNotificationHandler.java]: Method added to get the correct scenario.  Currently it just returns the single one we have for this journey.

**Review by jarekPierchala — APPROVED** [review]

**Pull request merged**